lab_1/main.py

```diff
@@ -87,16 +149,44 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
+    if (type(tokens) != list) or (type(word) != str):
+        return []
+    elif word not in tokens:
+        return []
+    elif (type(left_n) != int) and (type(right_n) != int):
+        return []
+    list_concordance = get_concordance(tokens, word, left_n, right_n)
+    adjacent_words = []
+    for context_list in list_concordance:
+        for index, token in enumerate(context_list):
+            if token == word:
+                if len(context_list[:index]) < left_n:
+                    if right_n < 1:
+                        adjacent_words += [[context_list[0]]]
+                    else:
+                        adjacent_words += [[context_list[0], context_list[index + right_n]]]
+                elif len(context_list[(index + 1):]) < right_n:
+                    if left_n < 1:
+                        adjacent_words += [[context_list[-1]]]
+                    else:
+                        adjacent_words += [[context_list[index - left_n], context_list[-1]]]
+                elif (type(left_n) != int) or (left_n < 1):
+                    adjacent_words += [[context_list[index + right_n]]]
+                elif (type(right_n) != int) or (right_n < 1):
+                    adjacent_words += [[context_list[index - left_n]]]
+                else:
+                    adjacent_words += [[context_list[index - left_n], context_list[index + right_n]]]
+
+    return adjacent_words
 
 
 def read_from_file(path_to_file: str) -> str:
     """
     Opens the file and reads its content
     :return: the initial text in string format
     """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    with open(path_to_file, 'r', encoding='utf-8') as file:
```

Comment:
It is usually considered a bad style (and a dangerous one!) to use built-in words (e.g. `str`, `list` etc.) as variable names.